src/libutil/file-system.cc

```diff
@@ -432,8 +394,15 @@ static void _deletePath(Descriptor parentfd, const Path & path, uint64_t & bytes
         AutoCloseDir dir(fdopendir(fd));
         if (!dir)
             throw SysError("opening directory '%1%'", path);
-        for (auto & i : readDirectory(dir.get(), path))
-            _deletePath(dirfd(dir.get()), path + "/" + i.name, bytesFreed);
+
+        struct dirent * dirent;
+        while (errno = 0, dirent = readdir(dir.get())) { /* sic */
```

Comment:
What's the reason for not using `readDirectory()` or `fs::directory_iterator` here?

Reply:
@edolstra the fact that it is using `DIR *` from the file descriptor, which is beyond what `std::filesystem` supports. So I inlined the old code which is just needed for this case.